inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -1111,19 +1177,76 @@ else if (identifierKBObj.get() instanceof KBInstance) {
                     language.ifPresent(handle::setLanguage);
                 }
             }
-            else {
-                handle.setName(handle.getUiLabel());
-            }
             
-            if (description != null) {
+            if (description != null ) {
                 handle.setDescription(description.getValue().stringValue());
             }
-            
             handles.add(handle);
+
         }
         return handles;
     }
 
+    /**
+     * Method process the Tuple Query Results
+     * @param kb KnowledgeBase variable
+     * @param tupleQuery Tuple Query Variable
+     * @param aAll True if entities with implicit namespaces (e.g. defined by RDF)
+     * @param itemVariable The variable to define the item IRI (eg.'s')
+     * @param langVariable The variable to define the item IRI (In general: 'l')
+     * @param descVariable The variable to define the item IRI (In general: 'd')
+     * @return list of all the {@link KBHandle} 
+     * @throws QueryEvaluationException
+     */
+    private Optional<KBHandle> evaluateGenericLabelQuery(KnowledgeBase aKB, TupleQuery tupleQuery,
+            boolean aAll, String itemVariable, String langVariable,
+            String descVariable)
+        throws QueryEvaluationException
+    {
+        TupleQueryResult result = tupleQuery.evaluate();        
+        
+        Optional<KBHandle> handleValue =  Optional.of(new KBHandle());
+        while (result.hasNext()) {
+            BindingSet bindings = result.next();
+            if (bindings.size() == 0) {
+                return Optional.empty();
+            }
+            String id = bindings.getBinding(itemVariable).getValue().stringValue();
+
+            if (!id.contains(":") || (!aAll && hasImplicitNamespace(id))) {
+                continue;
+            }
+
+            // Bindings without language specifications
+            Binding labelGeneral = bindings.getBinding("lGen");
+            Binding descGeneral = bindings.getBinding("dGen");
+            
+            KBHandle handle = new KBHandle(id);
+            
+            if (labelGeneral != null) {
+                handle.setName(labelGeneral.getValue().stringValue());
+                if (labelGeneral.getValue() instanceof Literal) {
+                    Literal literal = (Literal) labelGeneral.getValue();
+                    Optional<String> language = literal.getLanguage();
+                    language.ifPresent(handle::setLanguage);
+                }
+            }
+            
+            // needs to be fixed
```

Comment:
What needs to be fixed here?

Reply:
Nothing. Its an old comment when I was working on it. Will remove this.